runner/context.go

```diff
@@ -61,14 +61,27 @@ func (c *ContainerContext) WithVolumes(vols ...string) *ContainerContext {
 // 	return c
 // }
 
-func (c *ContainerContext) ParseContainerArgs(cargs []string) *ContainerContext {
-	c.parseUserArgs(cargs)
-	c.parseVolumes(cargs)
-	return c
+func (c *ContainerContext) ParseContainerArgs(cargs []string) (*ContainerContext, error) {
+	cargs, err := c.parseUserArgs(cargs)
+
+	if err != nil {
+		return nil, err
+	}
+
+	cargs = c.parseVolumes(cargs)
```

Comment:
you can rewrite this to be more concise like so

```
if _, err := c.parseRemaining(c.parseVolumes(cargs)); err != nil {
	return nil, err
}
return c, nil
```

Reply:
Yeah I wondered if that's a bit ugly, but can do

Reply:
What if we had more at some point, then it'd start to get very ugly fast?

Reply:
change the impementation slightly with an own struct for containerArgs parsing and management, in the future this should be extended more easily hopefully 🤞

Reply:
You have?